src/scan/saves.rs

```diff
@@ -20,11 +21,13 @@ pub struct ScannedFile {
     /// An enclosing archive file, if any, depending on the `BackupFormat`.
     pub container: Option<StrictPath>,
     pub redirected: Option<StrictPath>,
+    /// Optional. The store the file originated from.
+    pub store: Option<Store>,
```

Comment:
Technically, this should be a `HashSet<Store>`, since one path could be found in multiple roots. Let's say someone has their Epic and Amazon games installed together in `D:\Games`, so they make two roots pointing there. I think the current code would insert two `ScannedFile`s that are the same except for the store, and that might lead to a duplicate path in the results. `ScannedFile`s are tracked by their overall hash, not just the hash of the path (I should refactor that eventually 😅).

I see two options:

* Refactor `fn scan_game_for_backup` to change the local variable `found_files` from a `HashSet<ScannedFile>` to a `HashMap<StrictPath, ScannedFile>`. If a path is already in the map, then just add the store to it. We'd still convert it into a `HashSet` before we return it as part of the `ScanInfo`, to minimize the impact for now.
* Since we're not doing any file-level filtering anyway, we could put the `HashSet<Store>` in `ScanInfo` directly. In `fn scan_game_for_backup`, as we check each path,  we can just build up a hash set of roots with matches and then put that in the final results.

The first option gives us more info if we want to use it in the future. The second option keeps things simpler and just gives us what we need right now. I'm fine with either one you feel like doing, although I'd probably vote for the second option if I had to pick.